modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/cache/GridCacheCommandHandler.java

```diff
@@ -180,6 +180,12 @@
         CACHE_UPDATE_TLL
     );
 
+    /** Cache requests that can have null cache name {@code cacheName}. */
+    private static final EnumSet<GridRestCommand> CACHE_NAME_NOT_REQUIRED_REQUESTS = EnumSet.of(
+        CACHE_CLEAR,
```

Comment:
This looks dangerous actually. Imagine someone forgets to specify a cache name and removes all data in the grid accidentally. How about not allowing to call `clear` without a cache name in this same patch?

Reply:
Yes, I also paid attention to this, but CACHE_CLEAR is not part of the public REST API (https://apacheignite.readme.io/docs/rest-api#api-reference). Was not sure if it was used by some inner tool or by some customer, so just kept it.

Reply:
It is not documented, yes, but a lot of things are not documented, unfortunately.

I did a git annotate, and it seems CACHE_CLEAR was added as a part of a Redis command support here https://issues.apache.org/jira/browse/IGNITE-5706, but it is invokable on its own too.

Then, I looked into the usages of CACHE_CLEAR in the Redis protocol implementation. Based on what I see in GridRedisFlushCommandHandler, Redis always specifies the caches explicitly. It leads me to thinking that it should be safe to disallow CACHE_CLEAR without a cache name.

Reply:
Fixed that. CACHE_CLEAR remained in CACHE_NAME_NOT_REQUIRED_REQUESTS because it uses "values" field. "CacheName" is just ignored. By the way, corresponding test didn't not know that and was working only because it cleared all caches.